### ISSUE TITLE
Old results are 2 hours off 

### ISSUE THREAD

**Issue by svenvg93** [alexjustesen/speedtest-tracker#1487]
**Describe the bug**
After updating to v0.20.2 the old result from v0.19.0 are two hours off in time 

**To Reproduce**
1. Run speediest with 0.19
2. upgrade to 0.20.2 with alll the env needed
3. Check the results table

**Expected behavior**
The time should be correct.

**Environment (please complete the following information):**
 - OS: Ubuntu 24.04
-  Architecture: amd64
 - Browser Safari and Chrome
 - Version 0.20.2

Docker-compose:
```
services:
    speedtest-tracker:
        image: lscr.io/linuxserver/speedtest-tracker:0.20.2
        container_name: speedtest-tracker
        restart: unless-stopped
        environment:
            - TZ=Europe/Amsterdam 
            - APP_TIMEZONE=Europe/Amsterdam 
            - DISPLAY_TIMEZONE=Europe/Amsterdam 
            - PUID=1000
            - PGID=1000
            - DB_CONNECTION=sqlite
            - APP_KEY=base64:longkey
            - APP_URL=exmaple.com
            - SPEEDTEST_SCHEDULE=6 */2 * * *
            - SPEEDTEST_SERVERS=52365
            - PRUNE_RESULTS_OLDER_THAN=0
            - CHART_DATETIME_FORMAT="j M Y,  G:i:s"
        volumes:
            - speedtest-tracker:/config
        labels:
            - "traefik.enable=true"
            - "traefik.http.routers.speedtest.rule=Host(`example.com`)"
            - "traefik.http.routers.speedtest.entrypoints=websecure"
            - "traefik.http.routers.speedtest.tls=true"
            - "traefik.http.routers.speedtest.tls.certresolver=letencrypt"
            - "traefik.http.services.speedtest.loadbalancer.server.port=80"
        networks:
            - proxy

volumes:
  speedtest-tracker:
    name: speedtest-tracker

networks:
  proxy:
    name: proxy

```

**Screenshots**
v0.19

<img width="1222" alt="Scherm­afbeelding 2024-06-08 om 06 06 53" src="https://github.com/alexjustesen/speedtest-tracker/assets/4511676/55628f29-dbd8-404e-b0c3-18b00cb6dc8e">

v0.20

<img width="1233" alt="Scherm­afbeelding 2024-06-08 om 06 13 45" src="https://github.com/alexjustesen/speedtest-tracker/assets/4511676/a079aed2-de3e-42db-aa40-775d2dd5914e">

**Comment by AleksCee** [comment]
Had the same issue, you must have the same TZ in DB container and speed-test to work correct for Germany this works for me:

In speed-container:
            - TZ=Europe/Berlin
            - DATETIME_FORMAT="d.m.Y H:i:s"
            - CHART_DATETIME_FORMAT="d.m. H:i"
            - APP_TIMEZONE=Europe/Berlin
            - DISPLAY_TIMEZONE=Europe/Berlin
            
In Mariadb-Container:
- TZ=Europe/Berlin

**Comment by edd080** [comment]
I seem to have the same issue here too, how do you access the maria-db container please? It is the only part I did not check.
Thank you .

**Comment by svenvg93** [comment]
> Had the same issue, you must have the same TZ in DB container and speed-test to work correct for Germany this works for me:
> 
> In speed-container:
> 
> ```
>           - TZ=Europe/Berlin
>           - DATETIME_FORMAT="d.m.Y H:i:s"
>           - CHART_DATETIME_FORMAT="d.m. H:i"
>           - APP_TIMEZONE=Europe/Berlin
>           - DISPLAY_TIMEZONE=Europe/Berlin
> ```
> 
> In Mariadb-Container: `- TZ=Europe/Berlin`

Im not using an external DB, just SQLite. 
Fixed it by removing the `APP_TIMEZONE` from the environment variables. I missed understood the docs

**Comment by svenvg93** [comment]
Nevermind now the old ones are good, but the new results are 2 hours a head

**Comment by edd080** [comment]
I removed the app_timezone variable and it seemed to do the trick in my case :)
When running a new speedtest the results are showing the correct time too.

**Comment by svenvg93** [comment]
If I remove the APP_TIMEZONE the new results are two hours behind. If I remove DISPLAY_TIMEZONE the new results are 2 hours ahead.. 

As I had to rebuild everything yesterday before upgrading, the old results not much, so I can life the with the wrong time on them. But for other with more history it might be a problem.

**Comment by OddSquirrel** [comment]
Same here. Before updating from 19.0 to 20.2, I added the necessary environment variables to my Portainer stack. As for time-related settings, I added

```
- APP_TIMEZONE=Europe/Berlin
- DISPLAY_TIMEZONE=Europe/Berlin
```
to my speedtest-tracker container. Upon restart, I noticed that the time on the dashboard is correct, but the one shown in the results database is two hours behind. Then, I added

`- TZ=Europe/Berlin`

to my db container (PostgreSQL), but that didn't change anything—at least not for the data gathered so far.

Maybe the command isn't supposed to change anything about data already present, so I'll keep an eye on how it records the time stamps from now on. Luckily, I'm not that far off UTC, so it won't make that much of a difference.

Has anybody else made sense of it yet?

**Comment by alexjustesen** [comment]
@svenvg93 the whole timezone thing has historically been a pain point...

Can you try dropping `TZ=Europe/Amsterdam` from your app config?

**Comment by OddSquirrel** [comment]
N.B.: Now, that the next scheduled speed test ran, I can see that everything is in order. It seems that adding the time zone just doesn't do anything for the past, but the test was due to run @ 1 p.m. and it did. It also lists 1 p.m. in the results page.

That means, I only lost a couple of hours, which I don't really mind.

TL;DR: Add TZ settings to both app and db container and everything should be correct for following runs.

**Comment by svenvg93** [comment]
@alexjustesen dropped `TZ=Europe/Amsterdam` without result. Like @OddSquirrel mentioned all the new ones are correct just the old are off. Not a big deal if it's a pain to fix. 

Out of curiosity why the need for all the timezones and not just use the TZ for everything?

**Comment by alexjustesen** [comment]
`TZ` doesn't do anything inside of the framework / date library (Carbon), the only way to change how the timestamps are written to the database is to hook into the core `APP_TIMEZONE` config.

**Comment by alexjustesen** [comment]
Going to close this out, looks like we're all set.
